src/Pact/Analyze/Feature.hs

```diff
@@ -0,0 +1,1003 @@
+      $ Fun
+        Nothing
+        [ ("x", a)
+        ]
+        a
+  ]
+doc FNaturalLogarithm = Doc
+  "ln"
+  InvAndProp
+  "Logarithm of integers and decimals base e."
+  [ let a = TyVar $ TypeVar "a"
+    in Usage
+      "(ln x)"
+      (Map.fromList [("a", OneOf [int, dec])])
+      $ Fun
+        Nothing
+        [ ("x", a)
+        ]
+        a
+  ]
+doc FExponential = Doc
+  "exp"
+  InvAndProp
+  "Exponential of integers and decimals. e raised to the integer or decimal `x`."
+  [ let a = TyVar $ TypeVar "a"
+    in Usage
+      "(exp x)"
+      (Map.fromList [("a", OneOf [int, dec])])
+      $ Fun
+        Nothing
+        [ ("x", a)
+        ]
+        a
+  ]
+doc FAbsoluteValue = Doc
+  "abs"
+  InvAndProp
+  "Absolute value of integers and decimals."
+  [ let a = TyVar $ TypeVar "a"
+    in Usage
+      "(abs x)"
+      (Map.fromList [("a", OneOf [int, dec])])
+      $ Fun
+        Nothing
+        [ ("x", a)
+        ]
+        a
+  ]
+doc FBankersRound = Doc
+  "round"
+  InvAndProp
+  "Banker's rounding value of decimal `x` as integer, or to `prec` precision as decimal."
+  [ Usage
+      "(round x)"
+      Map.empty
+      $ Fun
+        Nothing
+        [ ("x", TyCon dec)
+        ]
+        (TyCon int)
+  , Usage
+      "(round x)"
+      Map.empty
+      $ Fun
+        Nothing
+        [ ("x",    TyCon dec)
+        , ("prec", TyCon int)
+        ]
+        (TyCon int)
+  ]
+doc FCeilingRound = Doc
+  "ceiling"
+  InvAndProp
+  "Rounds the decimal `x` up to the next integer, or to `prec` precision as decimal."
+  [ Usage
+      "(ceiling x)"
+      Map.empty
+      $ Fun
+        Nothing
+        [ ("x", TyCon dec)]
+        (TyCon int)
+  , Usage
+      "(ceiling x)"
+      Map.empty
+      $ Fun
+        Nothing
+        [ ("x",    TyCon dec)
+        , ("prec", TyCon int)
+        ]
+        (TyCon int)
+  ]
+doc FFloorRound = Doc
+  "floor"
+  InvAndProp
+  "Rounds the decimal `x` down to the previous integer, or to `prec` precision as decimal."
+  [ Usage
+      "(floor x)"
+      Map.empty
+      $ Fun
+        Nothing
+        [ ("x", TyCon dec)]
+        (TyCon int)
+  , Usage
+      "(floor x)"
+      Map.empty
+      $ Fun
+        Nothing
+        [ ("x",    TyCon dec)
+        , ("prec", TyCon int)
+        ]
+        (TyCon int)
+  ]
+doc FModulus = Doc
+  "mod"
+  InvAndProp
+  "Integer modulus"
+  [ Usage
+      "(mod x y)"
+      Map.empty
+      $ Fun
+        Nothing
+        [ ("x", TyCon int)
+        , ("y", TyCon int)
+        ]
+        (TyCon int)
+  ]
+
+-- Logical operators
+
+doc FGreaterThan = Doc
+  ">"
+  InvAndProp
+  "True if `x` > `y`"
+  [ let a = TyVar $ TypeVar "a"
+    in Usage
+      "(> x y)"
+      (Map.fromList [("a", OneOf [int, dec])])
+      $ Fun
+        Nothing
+        [ ("x", a)
+        , ("y", a)
+        ]
+        (TyCon bool)
+  ]
+doc FLessThan = Doc
+  "<"
+  InvAndProp
+  "True if `x` < `y`"
+  [ let a = TyVar $ TypeVar "a"
```

Comment:
You could make this a toplevel instead of redefining it 27 times

Reply:
Yeah -- agreed that this is not ideal. I guess we should move to explicit exports. Without explicit exports, this would require clients to do `hiding (a, ...)` or using a more verbose name instead of `a`